src/lib.rs

```diff
@@ -19,108 +19,201 @@ fn pivot(lower: usize, upper: usize) -> usize {
     return upper + ((lower - upper) / 2);
 }
 
-pub struct LazySortIterator<T, F> {
-    data: Vec<T>,
-    work: Vec<(usize, usize)>,
-    by: F,
+#[inline(always)]
+unsafe fn cmp_by<F, T>(by: &F, data: &mut [T], a: usize, b: usize) -> Ordering
+where
+    F: Fn(&T, &T) -> Ordering,
+{
```

Comment:
I'd recommend a debug assertions before every get unchecked. The the test suite helps find bugs, if there are.

Reply:
Yes, this is a good idea.